cdxev/build_public_bom.py

```diff
@@ -39,8 +39,18 @@ def remove_internal_information_from_properties(component: dict) -> None:
         component["properties"] = new_properties
 
 
+def process_component(component: dict[str, Any]) -> t.List[dict]:
+    cleared_components = []
+    remove_internal_information_from_properties(component)
+    cleared_components.append(component)
+    sub_components = extract_components(component.get("components", []))
+    for sub_component in sub_components:
+        process_component(sub_component)
+    return cleared_components
```

Comment:
The list you're returning is guaranteed to have exactly one element: `component`. If that was your intention, please make it clearer by returning `[component]`. Though in that case I would wonder why you're returning a list and not a dict.